src/pixel.rs

```diff
@@ -43,13 +48,13 @@ impl Grayscale {
         let alpha = reader.byte()?;
         Ok(Self { value, alpha })
     }
-    pub(crate) fn as_rgba(&self) -> Rgba {
+    pub(crate) fn to_rgba(self) -> Rgba {
```

Comment:
Why did you rename this function? Is it because it consumes the Self type? Maybe `into_rgba` would be clearer, as that parallels `into_iter`? Though, I'm not aware of a clear convention.

Reply:
It looks like "into" is the correct one in this case: https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv